File: ada/ada2012/Python3/Symbol.py
class Symbol:
    def __init__(self):
        self.name = ""
        self.classification = set()
        self.members = {}
        self.parent = None
        self.predefined = False
        self.is_composite = False
        self.defined_file = ""
        self.defined_line = 0
        self.defined_column = 0

    def __str__(self):
        result = self.name
        result += " (with classification " + ", ".join(str(c) for c in self.classification) + ")"
        if self.is_composite:
            result += " [composite]"
        if self.defined_file:
            result += f" defined at {self.defined_file}:{self.defined_line}:{self.defined_column}"
        return result

File: ada/ada2012/JavaScript/Symbol.js
export default class Symbol {
    constructor() {
        this.name = "";
        this.classification = new Set();
        this.members = new Map();
        this.parent = null;
        this.predefined = false;
        this.isComposite = false;
        this.definedFile = "";
        this.definedLine = 0;
        this.definedColumn = 0;
    }
    toString() {
        let result = this.name;
        result += " (with classification " + Array.from(this.classification).join(", ") + ")";
        if (this.isComposite) result += " [composite]";
        if (this.definedFile) result += " defined at " + this.definedFile + ":" + this.definedLine + ":" + this.definedColumn;
        return result;
    }
}
